app/services/invoices/create_service.rb

```diff
@@ -59,6 +59,19 @@ def from_date
                      (Time.zone.at(timestamp) - 1.year).to_date
                    else
                      raise NotImplementedError
+                   end
+
+      # NOTE: In case we are terminating old plan (paying in arrear),
+      # we should move to the beginning of the billing period
+      if subscription.terminated?
+        @from_date = case subscription.plan.interval.to_sym
+                     when :monthly
+                       Time.zone.at(timestamp).to_date.beginning_of_month
```

Comment:
we already covered the case If it's less than started_at (below)

Reply:
What do you think of moving this logic into a dedicated method?
We start to have a lot of logic in this one

Reply:
agree!